### PR TITLE
Blur fix

### LINKED ISSUE #2102: `transform.box_blur` works incorrectly.
**Test Code**
```python
import pygame
import sys

sf=pygame.image.load("peppers3.tif")
pygame.display.init()
dsf=pygame.display.set_mode((1280,720))

while 1:
    dsf.fill((0,150,0))

    dsf.blit(sf, (640,0,0,0))

    pygame.transform.box_blur(dsf,10,dest_surface=dsf)

    dsf.blit(sf,(0,0,1,1))
    pygame.display.update()
    for event in pygame.event.get():
        if event.type==pygame.QUIT:
            sys.exit()

```
image

### PR DESCRIPTION
**Pull request by yunline**
fix #2102
image

### CODE REVIEW

**Review by oddbookworm — APPROVED** [review]
LGTM! Fixes the linked issue nicely

File: src_c/transform.c [hunk 1]
```diff
@@ -3031,7 +3037,13 @@ box_blur(SDL_Surface *src, SDL_Surface *dst, int radius, SDL_bool repeat)
             }
         }
     }
-
+    if (inplace) {
+        for (i = 0; i < src->h; i++) {
+            memcpy(srcpx + i * src_pitch, dstpx + i * dst_pitch,
+                   sizeof(Uint8) * MIN(dst_pitch, src_pitch));
+        }
```
Comment (ankith26) [inline src_c/transform.c:3044]: Hmmm, instead of manual surface copying should we use something like `SDL_BlitSurface` instead?

Comment (yunline) [reply src_c/transform.c:3044]: The Surface `src` is locked and `SDL_BlitSurface` returns an error. I think copy manually may be better (?

Comment (Starbuck5) [reply src_c/transform.c:3044]: I wonder if surface pitches could be negative?

The sizeof(Uint8) is unnecessary.

If you used SDL_CreateRGBSurfaceWithFormatFrom in the setup you could make sure the new dst has the same pitch as the source. (You'd need to malloc the pixels yourself though, `src->BytesPerPixel * src->h  * src->pitch` worth. 
The advantage of making sure the pitch matches is that you could do a single malloc to copy the whole pixel region into the src/dest surface.

**Comment by yunline** [comment]
Solved another problem when use gaussian_blur in place.
I think it can be a part of this PR.
```py
import pygame
import sys

sf=pygame.image.load("peppers3.tif")
pygame.display.init()
dsf=pygame.display.set_mode((1280,720))
ddsf=dsf.subsurface((720,100,200,300))

while 1:
    dsf.fill((0,150,0))

    dsf.blit(sf, (640,0,0,0))

    pygame.transform.gaussian_blur(ddsf,50,dest_surface=ddsf)

    #dsf.blit(sf,(0,0,1,1))
    pygame.display.update()
    for event in pygame.event.get():
        if event.type==pygame.QUIT:
            sys.exit()
```
**Before fix**
image

**After fix**
image

**Comment by yunline** [comment]
It might be better to raise an error instead of creating a buffer when handling in-place operation.
People might think in-place operation is faster, but it's actually slower.

**Comment by Starbuck5** [comment]
>  It might be better to raise an error instead of creating a buffer when handling in-place operation.
People might think in-place operation is faster, but it's actually slower.

So an error if the source and destination surfaces are the same? I'm totally okay with that.

**Comment by yunline** [comment]
> So an error if the source and destination surfaces are the same? I'm totally okay with that.

OK, I'll open another PR for this

**Pull request closed without merging**